extensions/remote-udp/README.md

```diff
@@ -0,0 +1,67 @@
+# Remote UDP Extension
+
+The openEO API is a specification for interoperable cloud-based processing of large Earth observation datasets.
+
+This extension enables user to load user-defined processes that are hosted external to the openEO API (e.g., GitHub or cloud storage) through the process namespace into process graphs.
+
+- Version: **0.1.0**
+- Stability: **experimental**
+- Conformance class: `https://api.openeo.org/extensions/remote-udp/0.1.0`
+
+## Justification
+
+The openEO API defines the `namespace` property in a process node of a process graph as follows:
+>  The following options are predefined by the openEO API, but additional namespaces may be introduced by back-ends or in a future version of the API.
+> * `null` [...]
+> * `backend` [...]
+> * `user` [...]
+
+This makes it possible for this extension to add additional allowed values to the `namespace` property.
+
+## Specification
+
+This extension extends the `namespace` property of process graph nodes so that it accepts **absolute** URL with the protocols `https` (**recommended**) and `http` (discouraged). The URLs specified MUST return one of the following two options:
+
+1. A single process, compatible to the endpoint `GET /process_graphs/{process_graph_id}`.
+   In this case, the `id` property of the process graph node MUST be equal to the `id` of the process, 
+   otherwise a `ProcessNamespaceInvalid` error is thrown
+2. A list of processes, compatible to the endpoint `GET /process_graphs`.
+   In this case, the `id` property of the process graph node is used to identify the process from the list.
+   If not found a `ProcessNamespaceInvalid` error is thrown
+
+### Compatibility
+
+Compatible means in this context that the requests and responses must comply to the openEO API specification with the following exceptions:
+
+- The `Authorization` header MUST NOT not be sent.
+- Lists of processes MUST NOT paginate and the full process description MUST be provided for each process (i.e., the recommendation to omit large properties such as `process_graph` doesn't apply).
+
+### Client Considerations
+
+Clients MUST only offer this functionality to users if the conformance class of this extension is listed in the `conformsTo` property of the `GET /` endpoint.
```

Comment:
Maybe this is too much nitpicking, but can the "MUST" be relaxed to "SHOULD"? 
There are situations where you are building a process graph without knowing what the actual backend will be (e.g. because you are building generic UDPs), so there is no `GET /` for the client to check

Reply:
Makes sense and I think we can make it even more clear.

> If conncected to a specific backend, clients MUST only ...

This means you can check `GET /` and in all other cases you can offer it, of course (although I believe most or all clients right now create a connection first, otherwise you may also not have a list of processes to work with).

Reply:
> although I believe most or all clients right now create a connection first, 

we have some internal uses of the python client where we build process graphs without a  concrete connection.

And apart from that, when building non-concrete UDPs, you should also consider that there can be a considerable difference (in time) between the moment you build the UDP and the moment you execute/use the UDP, so it doesn't make sense to be very picky about the backend capabilities at build time when they only matter at execution time.

Reply:
Okay. Do you think this needs further fine-tuning?